### ISSUE TITLE
WP All Import download fails.

### ISSUE THREAD

**Issue by jmotes** [junaidbhura/composer-wp-pro-plugins#64]
See error in screenshot. This just started happening today - not sure why. 

I looked at the source code and tried to reconstruct the download URL like so:

~~https://www.wpallimport.com/?edd_action=version&license=[MY_LICENSE_KEY]&item_name=WP%20All%20Import&url=https%3A%2F%2Ftest.com%2F&version=4.7.9~~

~~That didn't work so I'm guessing that URL has changed unless I'm doing something wrong.~~

EDIT: I had an issue with the URL. This one works:
https://www.wpallimport.com/?edd_action=get_version&license=[MY_LICENSE_KEY]&item_name=WP%20All%20Import&url=https%3A%2F%2Ftest.com%2F&version=4.7.9

I'm not sure what could be the issue now. If it matters, I'm using Composer 2.4.2.

![image](https://github.com/junaidbhura/composer-wp-pro-plugins/assets/1943254/f3863b63-acab-4907-8d40-93a4d45ee6e6)

**Comment by mattneal-stafflink** [comment]
I've got the exact same issue atm.

**Comment by mcaskill** [comment]
Hrm. It must be an issue with WPAI's server if `extractDownloadUrl()` is receiving `null`.

`null` is returned if the JSON cannot be decoded or if the encoded data is deeper than the nesting limit.

```php
$response = json_decode( $http->get( 'https://www.wpallimport.com', array(
	'edd_action' => 'get_version',
	'license'    => $license,
	'item_name'  => $name,
	'url'        => $url,
	'version'    => $this->version,
) ), true );

return $this->extractDownloadUrl( $response );
```

~~This portion could benefit from some extra logic to intercept the failed decoding and throw a relevant message.~~

Ha. I completely forgot I had thought of this and have addressed in #58 that I proposed in March but don't have time to test and finish.

**Comment by jmotes** [comment]
Thought I'd link to this old ticket I just remembered from last year. Wonder if it's the same issue again: https://github.com/junaidbhura/composer-wp-pro-plugins/issues/41

**Comment by jmotes** [comment]
It's been working again for me today. It working for everyone else too now?

**Comment by junaidbhura** [comment]
WP All Import are notoriously unstable. Is it working for everyone again today?

**Comment by mcaskill** [comment]
Indeed. WPAI appears to be the most common issue. We should add a specific notice about it in both the README and in the error messages reported by Composer.

**Comment by justlevine** [comment]
I also started running into this 2 weeks ago. Interestingly, it was only happening on my local machine and not my CI.

Copying over the changes to the `Http` class made in #58 showed me the issue wasnt actually with WPAI (for a change), but `cURL error (60): SSL certificate problem: unable to get local issuer certificate`, and updating the `wp-includes/certificates/ca-bundle.crt` fixed it for me.

**Comment by mcaskill** [comment]
> `cURL error (60): SSL certificate problem: unable to get local issuer certificate`

That's a good find and something the Composer plugin could try to check for in the HTTP response to report to the user.

**Comment by jmotes** [comment]
Having problems again today 😓

**Comment by justlevine** [comment]
@jmotes I think we can probably assume at this point that all wpai errors are caused by their servers and not this repo.

**Comment by jmotes** [comment]
@justlevine Yeah - it definitely seems to be Cloudflare blocking us again. Since I'm using Azure Devops for CI/CD I decided to just upload all of the WPAI suite plugins to the project's secure files library and unarchive them at build time to create my deployment artifact. It took a couple hours out of my morning to get it all working but happy to have something a bit more failproof without committing them directly to the repo.

It stinks that I have to re-upload the plugins every time I want to update to the latest version (which this project handles nicely) but at least I can continue using it for Gravity Forms and ACF.

**Comment by junaidbhura** [comment]
Hey everyone, please share your thoughts here: https://github.com/junaidbhura/composer-wp-pro-plugins/issues/69

**Comment by Twansparant** [comment]
Jup, same error here since yesterday...

**Comment by nikolowry** [comment]
It be nice to drop https://github.com/junaidbhura/composer-wp-pro-plugins/blob/master/src/Plugins/AbstractEddPlugin.php#L39-L46 for the WPAI plugins -- or at least offer some type of ENV/flag to skip that check entirely.

My CI/CD builds are failing because the version got pinned in `composer.lock`, even though I've specified `*` for the version in my `composer.json`.

**Comment by rodrigo-arias** [comment]
I'm getting a similar error. It works fine locally but fails in CI. I assume it's an issue with the WPAI servers since I haven't made any significant changes since the last time it worked, aside from updating the version to 4.9.3.

![CleanShot 2024-11-08 at 10 19 59](https://github.com/user-attachments/assets/2f603bfd-7e10-40ef-8d3b-592c2c4b0fd6)

```
{
	"type": "package",
	"package": {
		"name": "junaidbhura/wp-all-import-pro",
		"version": "4.9.3",
		"type": "wordpress-plugin",
		"dist": {
			"type": "zip",
			"url": "https://www.wpallimport.com/"
		},
		"require": {
			"junaidbhura/composer-wp-pro-plugins": "*"
		}
	}
}
```
